daemon/tests/feedsDatabase/feedsDatabase-tests.ts

```diff
@@ -0,0 +1,89 @@
+import { expect } from 'chai';
+import fs from 'fs';
+import * as mocha from 'mocha'; // import types for mocha e.g. describe
+import os from 'os';
+import { createKeyPair, publicKeyHexToFeedId, publicKeyToHex } from '../../src/common/crypto_util';
+import { randomAlphaString } from '../../src/common/util';
+import LocalFeedsDatabase from '../../src/external/real/LocalFeedsDatabase';
+import { isSignature, localFilePath, nowTimestamp, SignedSubfeedMessage, SubfeedAccessRules, SubfeedHash, SubfeedMessage } from '../../src/interfaces/core';
+
+const testContext = (testFunction: (localFeedsDatabase: LocalFeedsDatabase, resolve: () => void, reject: (err: Error) => void) => Promise<void>, done: (err?: Error) => void) => {
+    const tempPath = `${os.tmpdir()}/kachery-p2p-test-${randomAlphaString(10)}.tmp`
+    fs.mkdirSync(tempPath.toString())
+    const resolve = () => {
+        fs.rmdirSync(tempPath, {recursive: true})
+        done()
+    }
+    const reject = (err: Error) => {
+        fs.rmdirSync(tempPath, {recursive: true})
+        done(err)
+    }
+
+    const fd = new LocalFeedsDatabase(localFilePath(tempPath + '/feeds.db'))
+
+    testFunction(fd, resolve, reject).then(() => {
+    }).catch((err: Error) => {
+        reject(err)
+    })
+}
+
+// need to explicitly use mocha prefix once or the dependency gets wrongly cleaned up
+mocha.describe('Feeds database', () => {
```

Comment:
Noting that this is an integration test rather than a unit test.

Not blocking, but when we have the chance to expand it, it'd be nice to:
- add internal logic tests with mocks (unit tests)
- break this test into several (even though repeating the test fixtures is time-consuming) to test different functionality discretely
- test that failure modes work as expected--I see you're deleting the feed and rerunning some of the lookups, so maybe we're okay on this front?

Reply:
added this as comment in source.